tiny_dnn/activations/elu_layer.h

```diff
@@ -41,3 +41,8 @@ class elu_layer : public activation_layer {
 };
 
 }  // namespace tiny_dnn
+
+// allow upcasting with cereal
+#ifndef CNN_NO_SERIALIZATION
+CEREAL_REGISTER_TYPE(tiny_dnn::elu_layer);
```

Comment:
I think It would be better if you can move `CEREAL_REGISTER_TYPE(...)` macros into a single header file.
They are scattered now.